src/dsp_tools/utils/uri_util.py

```diff
@@ -12,3 +12,55 @@ def is_uri(s: str) -> bool:
     fragment = r"(?<fragment>#[\w_.\-~:/]*)"
     m = regex.match(rf"{scheme}:(//{host}{port}?){path}*{query}*{fragment}?", s, flags=regex.UNICODE)
     return m is not None
+
+
+def is_iiif_uri(uri: str) -> bool:
+    """
+    Checks if the given URL is a valid IIIF URL.
+    It should support all versions of IIIF servers,
+    but was constructed with the syntax described in: https://iiif.io/api/image/3.0/#2-uri-syntax
+    IIIF is open to modifications which are not described in the official documentation.
+    https://iiif.io/api/annex/notes/design_principles/#define-success-not-failure
+    This may need to be updated in case of a valid version which is not in the regex.
+
+    Args:
+        uri: The URI to be checked.
+
+    Returns:
+        True if the URI is a valid IIIF URI, False otherwise.
+    """
+    # {scheme}://{server}{/prefix}/{identifier}/{region}/{size}/{rotation}/{quality}.{format}
+    split_uri = uri.lower().split("/")
+    if len(split_uri) < 8:
+        return False
+    # %5E is the URL encoded version of ^ ->
+    # because we do change the uri to lower case we need to change that in the regex
+    # (\d+(\.\d+)?) -> number, can be integer or float
```

Comment:
To make it easier: store that in a variable, then the regexes below become easier